include/sampleflow/consumers/covariance_matrix.h

```diff
@@ -171,7 +171,7 @@ namespace SampleFlow
           delta -= current_mean;
           for (unsigned int i=0; i<sample.size(); ++i)
             for (unsigned int j=0; j<sample.size(); ++j)
-              current_covariance_matrix(i,j) += delta[i]*delta[j]/n_samples;
+              current_covariance_matrix(i,j) += delta[i]*delta[j]/n_samples-current_covariance_matrix(i,j)/(n_samples-1);
```

Comment:
This part of the patch is unrelated to everything else you do. It's often easiest to do this in a separate pull request.

Want me to extract this piece into something separate?

Reply:
Yes, sure. Not sure how that "separate" looks like.

Reply:
I can do that for you.

Reply:
Let's get everything else in order first, though.